freqtrade/data/dataprovider.py

```diff
@@ -63,9 +79,100 @@ def _set_cached_df(
         :param dataframe: analyzed dataframe
         :param candle_type: Any of the enum CandleType (must match trading mode!)
         """
-        self.__cached_pairs[(pair, timeframe, candle_type)] = (
+        pair_key = (pair, timeframe, candle_type)
+        self.__cached_pairs[pair_key] = (
             dataframe, datetime.now(timezone.utc))
 
+    def add_external_df(
+        self,
+        pair: str,
+        timeframe: str,
+        dataframe: DataFrame,
+        candle_type: CandleType,
+    ) -> None:
+        """
+        Add the pair data to this class from an external source.
+
+        :param pair: pair to get the data for
+        :param timeframe: Timeframe to get data for
+        :param candle_type: Any of the enum CandleType (must match trading mode!)
+        """
+        pair_key = (pair, timeframe, candle_type)
+
+        # For multiple leaders, if the data already exists, we'd merge
+        self.__external_pairs_df[pair_key] = (dataframe, datetime.now(timezone.utc))
+        self._set_data_event(pair_key)
+
+    def get_external_df(
+        self,
+        pair: str,
+        timeframe: str,
+        candle_type: CandleType
+    ) -> DataFrame:
+        """
+        Get the pair data from the external sources. Will wait if the policy is
+        set to, and data is not available.
+
+        :param pair: pair to get the data for
+        :param timeframe: Timeframe to get data for
+        :param candle_type: Any of the enum CandleType (must match trading mode!)
+        """
+        pair_key = (pair, timeframe, candle_type)
+
+        if pair_key not in self.__external_pairs_df:
+            self._wait_on_data(pair_key)
+
+            if pair_key not in self.__external_pairs_df:
+                return (DataFrame(), datetime.fromtimestamp(0, tz=timezone.utc))
+
+        return self.__external_pairs_df[pair_key]
```

Comment:
At this point, multiple leaders could be merged into a final DF that goes back to the strategy.

Reply:
i think for multiple leaders, you'd want the strategy to pull data for each leader it'd want to query.
that'll leave the "how" to the strategy.